lib/tasks/helpers/newrelicyml.rb

```diff
@@ -0,0 +1,142 @@
+      # Relic. For more details, see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/renaming-applications
+      app_name: <%= app_name %>
+
+      # To disable the agent regardless of other settings, uncomment the following:
+      # agent_enabled: false
+
+      # Logging level for log/newrelic_agent.log; options are error, warn, info, or
+      # debug.
+      log_level: info
+
+      # All of the following configuration options are optional. Review them, and
+      # uncomment or edit them if they appear relevant to your application needs.
+
+  HEADER
+
+  FOOTER = <<~FOOTER
+    # Environment-specific settings are in this section.
+    # RAILS_ENV or RACK_ENV (as appropriate) is used to determine the environment.
+    # If your application has other named environments, configure them here.
+    development:
+      <<: *default_settings
+      app_name: <%= app_name %> (Development)
+
+    test:
+      <<: *default_settings
+      # It doesn't make sense to report to New Relic from automated test runs.
+      monitor_mode: false
+
+    staging:
+      <<: *default_settings
+      app_name: <%= app_name %> (Staging)
+
+    production:
+      <<: *default_settings
+  FOOTER
+
+  def self.get_configs(defaults)
+    final_configs = {}
+
+    defaults.sort.each do |key, value|
+      next if CRITICAL.include?(key) || SKIP.include?(key)
+
+      if public_config?(value)
+        sanitized_description = sanitize_description(value[:description])
+        description = format_description(sanitized_description)
+        default = default_value(key, value)
+        final_configs[key] = {description: description, default: default}
+      end
+    end
+
+    final_configs
+  end
+
+  def self.public_config?(value)
+    value[:public] == true
+  end
+
+  def self.sanitize_description(description)
+    # remove callouts
+    description = description.split("\n").reject { |line| line.match?('</?Callout') }.join("\n")
+    # remove InlinePopover, keep the text inside type
+    description.gsub!(/<InlinePopover type="(.*)" \/>/, '\1')
+    # remove hyperlinks
+    description.gsub!(/\[([^\]]+)\]\([^\)]+\)/, '\1')
+    # remove single pairs of backticks
+    description.gsub!(/`([^`]+)`/, '\1')
+    # removed hrefs links
+    description.gsub!(/<a href="(.*)">(.*)<\/a>/, '\2')
+
+    description
+  end
+
+  def self.format_description(description)
+    # remove leading and trailing whitespace
+    description = description.strip
+    # wrap text after 80 characters
+    description.gsub!(/(.{1,80})(\s+|\Z)/, "\\1\n")
+    # add hashtags to lines
+    description = description.split("\n").map { |line| "  # #{line}" }.join("\n")
+
+    description
+  end
+
+  def self.default_value(key, config_hash)
+    if PROCS.include?(key)
+      PROCS[key]
+    else
+      default = config_hash[:documentation_default].nil? ? config_hash[:default] : config_hash[:documentation_default]
+      default = 'nil' if default.nil?
```

Comment:
Ah here's the `'nil'`. I'd like to know why this is needed over `nil` or `''`.

Reply:
We need the string version of `nil` and the string of double quotes`""` because we are putting these in the yml as strings. If I don't stringify these, the values get evaluated and then we end up with blank values. ie `default: ` vs `default: nil`